db/openshift.deploy.yml

```diff
@@ -326,3 +329,86 @@ objects:
+        spec:
+          backoffLimit: ${{JOB_BACKOFF_LIMIT}}
+          template:
+            metadata:
+              labels:
+                app: ${NAME}-${ZONE}
+                cronjob: ${NAME}-${ZONE}
+            spec:
+              containers:
+                - name: ${NAME}-${ZONE}-${COMPONENT}-restore
+                  # use the same image as our database, so we can run the psql command
+                  image: image-registry.apps.silver.devops.gov.bc.ca/${OC_NAMESPACE}/${NAME}-${ZONE}-${COMPONENT}:${ZONE}-db
+                  command: ["/bin/sh", "-c"]
+                  args:
+                  - |
+                    sql_zip_file=$(find backups/${RESTORE_DIR} -type f -name "*.sql.gz" -print0)
+                    echo "Found zipped backup file: $sql_zip_file"
+                    gzip -d $sql_zip_file
+                    sql_file=$(find backups/${RESTORE_DIR} -type f -name "*.sql" -print0)
+                    echo "Running SQL file: $sql_file"
+                    psql -h ${TARGET_HOST} -U ${POSTGRES_USER} -d ${POSTGRES_DB} -f $sql_file
+                    gzip $sql_file
+                    echo "Finish database restore"
+                  volumeMounts:
+                    - name: ${NAME}-${ZONE}-${COMPONENT}-restore
+                      mountPath: "${BACKUP_DIR}"
+                    - name: ${NAME}-${ZONE}-${COMPONENT}-restore
+                      mountPath: /var/lib/postgresql
+                  env:
+                    - name: BACKUP_DIR
+                      value: "${BACKUP_DIR}"
+                    - name: POSTGRES_DB
+                      valueFrom:
+                        secretKeyRef:
+                          name: ${NAME}-${ZONE}-${COMPONENT}
+                          key: database-db
+                    # this enviornment variable mmust be named PGPASSWORD, so postgres knows this is the password, and will not ask us to enter manually
+                    - name: PGPASSWORD
+                      valueFrom:
+                        secretKeyRef:
+                          name: ${NAME}-${ZONE}-${COMPONENT}
+                          key: database-password
+                    - name: POSTGRES_USER
+                      valueFrom:
+                        secretKeyRef:
+                          name: ${NAME}-${ZONE}-${COMPONENT}
+                          key: database-user
+                    # we don't need to declare the restore_dir variable if we set it manually, just put here so we won't forget
+                    # restore_dir need to be manually set, in the format for example, daily/2024-08-16, where backup container puts the backup files
+                    - name: RESTORE_DIR
+                      value: ""
+                    # we don't need to declare the target_host variable if we set it manually, just put here so we won't forget
+                    # target_host is the ip address of the new database pod
+                    - name: TARGET_HOST
```

Comment:
I don't get this comment - don't we have to set TARGET_HOST? Not sure why we're spinning up a new database pod in a restore scenario - how is the application being repointed then at the restored database?

Reply:
So we will need to manually setup these values in order to trigger this job. The target host needs to be the IP address of the database pod, which could change whenever the database redeploy.

That's also why I feel it's okay to have a cronjob. Even someone triggers the job, without setting up these values, the job will fail. 

Please ignore my spinning up new database scenario for now. I have another idea and need to explain that.